src/util.c

```diff
@@ -385,3 +384,35 @@ parse_key_value_string (char *kv_str,
 out:
     return rc;
 }
+
+#define ENTROPY_SRC "/dev/urandom"
+static void* ptr_mask = NULL;
+
+void
+util_init (void)
+{
+    g_assert (ptr_mask == NULL);
+    int ret;
+    Random *rand;
+
+    g_debug ("%s", __func__);
+    rand = random_new ();
+    ret = random_seed_from_file (rand, ENTROPY_SRC);
+    g_assert (ret != -1);
+
+    g_debug ("%s: ptr_mask: %p", __func__, ptr_mask);
+    ret = random_get_bytes (rand, (uint8_t*)&ptr_mask, sizeof (ptr_mask));
+    g_assert (ret == sizeof (ptr_mask));
+    g_debug ("%s: ret: %d, ptr_mask: %p", __func__, ret, ptr_mask);
+
+    g_clear_object (&rand);
+}
+
+void*
+objid (const void *ptr)
+{
+    g_debug ("%s: ptr_mask: %p", __func__, ptr_mask);
+    g_assert (ptr_mask != NULL);
+
+    return (void*)((uintptr_t)ptr_mask ^ (uintptr_t)ptr);
```

Comment:
I guess in this case, the XOR Cipher is fine since key length == message length and the plain text is fully controlled. But I am no crypto expert.

Reply:
If you print a NULL pointer you print the key....

Reply:
Since this never happens in the code I didn't consider it an issue. Easy enough to have the `objid` function balk if passed a `NULL` pointer. Will keep me from screwing things up in the future.

Reply:
w/r/t use of XOR in your first comment: I looked into your initial example of using the kernel and it's `hash` function to mask the address. The built-in `hash` functions from the kernel aren't "cryptographic hash functions" so they're only intended for use in hash tables and the like. Relying on the "one-way" property of these hashes is a "very bad thing".

XOR seemed like a better solution for the reasons you list above. Alternatively using a cryptographic hash is possible but it will significantly increase the size of the output (from sizeof (void*) to 128 bytes in the case of SHA2) and it's more expensive.

Reply:
I think you can just return NULL if passed NULL since NULL is never really a usable info leak.

> XOR seemed like a better solution for the reasons you list above. Alternatively using a cryptographic hash is possible but it will significantly increase the size of the output (from sizeof (void*) to 128 bytes in the case of SHA2) and it's more expensive.

It's only expensive on the debug case. I think the xor cipher is fine here, again I am not nor claim to be a crypto expert.